### ISSUE TITLE
Create a system for deriving a pmf model from a generic base model.

### ISSUE THREAD

**Issue by hol353** [APSIMInitiative/ApsimX#7384]
There are a lot of plant models in APSIM that are (or should be) very similar to other models. e.g. barley and oats should be similar to wheat. At the moment we don't know what differences there are between wheat/barley/oats or between the different legume models. Some consistency is needed.

We would like to create the concept of a base (generic) model that other models can derive from. e.g. we could create a winter cereal model that wheat, barley and oats derive from. 

One way to implement this is to use the property overriding system that cultivars and factorials use. A plant model could then be the name of the base plant model and 1 or more property overrides. This would restrict derived models to simple re-parameterisation and not allow different model structures, which is one of the objectives for doing this. We already have code in APSIM to do parameter setting so the task should be straight-forward, although some refactoring will be needed first.

**Comment by EnliWang** [comment]
I find this is great and would like strongly support this approach.

What happens when new model components/approaches/structures are needed? Do we implement them in the base model? Or does the base model features polymorphism?

**Comment by jbrider** [comment]
@hol353 I have made changes recently on a local fork to try and decouple the commands from needing to include all of the Apsim library to work.  We talk to Apsim through the sockets from external applications.  The external apps shouldn't need Apsim libraries to be able to send simple replacement changes to ApsimServer.  The main change I made was to decouple the Commands from their handling - leaving that to Apsim to do internally in response to simple messages which don't require Apsim libraries to be created and sent.

I haven't pushed it back up yet because it breaks existing code and it is very ugly having been created to get things working quickly. Can we get together to chat about this - it's probably going to need a whiteboard.
